File: 2020/src/days/Day01.java
package days;

import java.util.*;
import java.io.*;

import utilities.*;
import utilities.exceptions.SolutionNotFoundException;

public class Day01{

    private static int partOne(int[] input) throws SolutionNotFoundException{
        for(int i = 0; i < input.length - 1; i++){
            for(int j = i+1; j < input.length; j++){
                if(input[i] + input[j] == 2020){
                    return input[i] * input[j];    
                }  
            }
        }
        throw new SolutionNotFoundException("Solution not found for part one!");
    }

    private static int partTwo(int[] input) throws SolutionNotFoundException{
        for(int i = 0; i < input.length - 2; i++){
            for(int j = i+1; j < input.length - 1; j++){
                for(int n = j+1; n < input.length; n++){
                    if(input[i] + input[j] + input[n] == 2020){
                        return input[i] * input[j] * input[n];    
                    }
                }
            }
        }  
        throw new SolutionNotFoundException("Solution not found for part two!"); 
    }

    public static void main(String[] args) throws SolutionNotFoundException{

        Scanner sc = new Scanner(System.in);

        int[] input = Parser.getIntArray(sc);
        
        System.out.println(partOne(input));
        System.out.println(partTwo(input));
        
        sc.close();
    }
}

File: 2021/src/days/Day01.kt
package days

import utilities.readInts

class Day01 {

    fun partOne(input: List<Int>): Int {

        var result = 0

        for (i in 1 until input.size){
            if (input[i] > input[i-1]){
                result++
            }
        }

        return result
    }

    fun partTwo(input: List<Int>): Int {
        return input.windowed(4).count { it.last() > it.first() }
    }
}

fun main(){

    val dayOne = Day01()

    val input = readInts()

    println(dayOne.partOne(input))
    println(dayOne.partTwo(input))

}